[CI] HeapAttackIT testLookupExplosionManyMatches failing

**Build Scans:**
- [elasticsearch-periodic #6237 / single-processor-node-tests](https://gradle-enterprise.elastic.co/s/ogl43rnum7jrq)
- [elasticsearch-periodic-platform-support #6219 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/txjlqbgycg35m)
- [elasticsearch-periodic-platform-support #6212 / oraclelinux-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/mqxndll6qirh6)
- [elasticsearch-periodic-platform-support #6107 / windows-2022_checkpart1_platform-support-windows](https://gradle-enterprise.elastic.co/s/iup2ss7a2b6x6)

**Reproduction Line:**
```
./gradlew ":test:external-modules:test-esql-heap-attack:javaRestTest" --tests "org.elasticsearch.xpack.esql.heap_attack.HeapAttackIT.testLookupExplosionManyMatches" -Dtests.seed=318CEC8A8DAB5000 -Dtests.configure_test_clusters_with_one_processor=true -Dtests.locale=nds-NL -Dtests.timezone=Antarctica/Casey -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.heap_attack.HeapAttackIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testLookupExplosionManyMatches),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected a map containing
  status: expected <429> but was <400>
   error: a map containing
      type: expected "circuit_breaking_exception" but was "resource_already_exists_exception"
    reason: "index [sensor_lookup/VpS0dnT_RvySWWLP14cIqQ] already exists" unexpected but ok
index_uuid: "VpS0dnT_RvySWWLP14cIqQ" unexpected but ok
     index: "sensor_lookup" unexpected but ok
root_cause: <[{reason=index [sensor_lookup/VpS0dnT_RvySWWLP14cIqQ] already exists, index_uuid=VpS0dnT_RvySWWLP14cIqQ, index=sensor_lookup, type=resource_already_exists_exception}]> unexpected but ok
```

**Issue Reasons:**
- [main] 4 failures in test testLookupExplosionManyMatches (0.7% fail rate in 603 executions)
- [main] 3 failures in pipeline elasticsearch-periodic-platform-support (25.0% fail rate in 12 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/test/external-modules/esql-heap-attack/src/javaRestTest/java/org/elasticsearch/xpack/esql/heap_attack/HeapAttackIT.java
+++ b/test/external-modules/esql-heap-attack/src/javaRestTest/java/org/elasticsearch/xpack/esql/heap_attack/HeapAttackIT.java
@@ -684,6 +684,7 @@ public class HeapAttackIT extends ESRestTestCase {
             return responseAsMap(query(query.toString(), null));
         } finally {
             deleteIndex("sensor_data");
+            deleteIndex("sensor_lookup");
         }
     }
 

